monai/transforms/inverse.py

```diff
@@ -0,0 +1,121 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from enum import Enum
+from typing import Dict, Hashable, Optional, Tuple
+
+import numpy as np
+
+from monai.transforms.transform import RandomizableTransform, Transform
+
+__all__ = ["InvertibleTransform"]
+
+
+class InvertibleTransform(Transform):
+    """Classes for invertible transforms.
+
+    This class exists so that an ``invert`` method can be implemented. This allows, for
+    example, images to be cropped, rotated, padded, etc., during training and inference,
+    and after be returned to their original size before saving to file for comparison in
+    an external viewer.
+
+    When the `__call__` method is called, the transformation information for each key is
+    stored. If the transforms were applied to keys "image" and "label", there will be two
+    extra keys in the dictionary: "image_transforms" and "label_transforms". Each list
+    contains a list of the transforms applied to that key. When the ``inverse`` method is
+    called, the inverse is called on each key individually, which allows for different
+    parameters being passed to each label (e.g., different interpolation for image and
+    label).
+
+    When the ``inverse`` method is called, the inverse transforms are applied in a last-
+    in-first-out order. As the inverse is applied, its entry is removed from the list
+    detailing the applied transformations. That is to say that during the forward pass,
+    the list of applied transforms grows, and then during the inverse it shrinks back
+    down to an empty list.
+
+    The information in data[key_transform] will be compatible with the default collate
+    since it only stores strings, numbers and arrays.
+
+    We currently check that the id() of the transform is the same in the forward and
+    inverse directions. This is a useful check to ensure that the inverses are being
+    processed in the correct order. However, this may cause issues if the id() of the
+    object changes (such as multiprocessing on Windows). If you feel this issue affects
+    you, please raise a GitHub issue.
+
+    Note to developers: When converting a transform to an invertible transform, you need to:
+        1. Inherit from this class.
+        2. In `__call__`, add a call to `push_transform`.
+        3. Any extra information that might be needed for the inverse can be included with the
+            dictionary `extra_info`. This dictionary should have the same keys regardless of
+            whether `do_transform` was True or False and can only contain objects that are
+            accepted in pytorch's batch (e.g., `None` is not allowed).
+        4. Implement an `inverse` method. Make sure that after performing the inverse,
+            `remove_most_recent_transform` is called.
+    """
+
+    class Keys(Enum):
+        """Extra meta data keys used for inverse transforms."""
+
+        class_name = "class"
```

Comment:
I think it's better to use capital words for the keys, to be consistent with all the existing MONAI enums:
https://github.com/Project-MONAI/MONAI/blob/master/monai/utils/enums.py

Reply:
done